pkg/api/networkservice/mechanisms/kernel/helpers.go

```diff
@@ -120,20 +101,17 @@ func (m *mechanism) GetInterfaceName(conn *networkservice.Connection) string {
 	return name
 }
 
-// GetNetNSURL - returns the NetNS URL - fmt.Sprintf("inode://%d/%d",dev,ino)
-func (m *mechanism) GetNetNSURL() string {
-	if m == nil || m.GetParameters() == nil {
-		return ""
-	}
+// SetInterfaceName sets the Kernel Interface Name
+func (m *Mechanism) SetInterfaceName(interfaceName string) {
+	m.GetParameters()[InterfaceNameKey] = interfaceName
+}
+
+// GetNetNSURL returns the NetNS URL - fmt.Sprintf("inode://%d/%d",dev,ino)
+func (m *Mechanism) GetNetNSURL() string {
```

Comment:
Won't this panic if m == nil or GetParameters() == nil?

Reply:
Fixed, now GetParameters() returns empty map for m == nil.